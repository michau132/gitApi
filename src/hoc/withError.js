import React, { Fragment } from 'react';
import styled from 'styled-components';
import CloseBtn from '../components/CloseBtn';

const ErrorP = styled.p`
  color: red;
`;

const withError = Component => (props) => {
  const { error } = props;
  if (error) {
    return (
      <Fragment>
        <CloseBtn />
        <ErrorP>Ups something goes wrong</ErrorP>
      </Fragment>
    );
  }
  return (
    <Component {...props} />
  );
};

export default withError;
