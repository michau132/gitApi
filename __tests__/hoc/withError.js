import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import withError from '../../src/hoc/withError';

const component = () => <h1>Hi there</h1>;
const ConditionalHOC = withError(component);
const props = { error: true };

describe('testing withError HOC', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ConditionalHOC {...props} />);
  });
  test('should render component when loading is set to true', () => {
    expect(wrapper).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
