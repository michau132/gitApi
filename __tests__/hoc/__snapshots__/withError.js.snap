// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing withError HOC should match snapshot 1`] = `
<Fragment>
  <CloseBtn />
  <styled.p>
    Ups something goes wrong
  </styled.p>
</Fragment>
`;
